ideascube/utils.py

```diff
@@ -87,3 +88,7 @@ def get_file_sha256(path):
 
 def get_file_size(path):
     return os.stat(path).st_size
+
+def tag_splitter(tag_string):
+    tags = set(t.strip().lower() for t in re.split('[;,]+', tag_string) if t.strip())
```

Comment:
I'm not sure we should lower-case the tags like this. 😕 

It has the potential to be one of these "features" which feels strongly like a bug for users: "come on, I keep using a capital letter and this dumb thing lower-cases it!!!!" followed by cries of rage.

Can't we lower-case the slug, to avoid duplicates, but keep the visible value the exact way the user typed it?